src/jit/block.cpp

```diff
@@ -1393,15 +1393,16 @@ bool BasicBlock::hasEHFlowIn()
 }
 
 //------------------------------------------------------------------------
-// blockHasEHFlowOut: Determine if this block ends in exception flow.
+// blockhasEHBoundaryOut: Determine if this block ends in an EH boundary.
 //
 // Arguments:
 //    block   - the BasicBlock
 //
 // Return Value:
-//    True iff the block ends in exception flow; false otherwise
+//    True iff the block end in an exception boundary that requires that all lclVars
+//    be valid on the stack; false otherwise.
 //
-bool BasicBlock::hasEHFlowOut()
+bool BasicBlock::hasEHBoundaryOut()
 {
     // If a predecessor is marked BBF_KEEP_BBJ_ALWAYS, then we must keep all live incoming
     // vars on the stack.
```

Comment:
I think it would be nice to split these conditions (if the comment is only the first case):
```
//  comment
if (((bbFlags & BBF_KEEP_BBJ_ALWAYS) != 0)
{
  return true;
}

if (bbJumpKind == BBJ_EHFILTERRET)
{
  return true;
}

if (bbJumpKind == BBJ_EHFINALLYRET)
{
  return true;
}

```